src/gui/gui_map_icons.hpp

```diff
@@ -580,12 +584,23 @@ class tl_status : public image_element_base {
 	}
 };
 
-class tl_strength : public color_text_element {
+class tl_strength : public multiline_text_element_base {
 public:
 	void on_update(sys::state& state) noexcept override {
 		top_display_parameters* params = retrieve<top_display_parameters*>(state, parent);
-		color = text::text_color::gold;
-		set_text(state, text::format_float(params->top_left_value, 1));
+		auto strength = params->top_left_value;
+		bool is_ship = float(int32_t(strength)) == strength;
+		auto layout = text::create_endless_layout(internal_layout,
+		text::layout_parameters{0, 0, int16_t(base_data.size.x), int16_t(base_data.size.y), base_data.data.text.font_handle, 0, text::alignment::center, text::text_color::gold, false});
+		auto box = text::open_layout_box(layout, 0);
+
+
+		text::add_to_layout_box(state, layout, box, text::format_float(strength, 1), text::text_color::gold);
+		if(!is_ship)
```

Comment:
You still have the integer test for whether things are ships, which won't work for reasons we have discussed